app/src/main/java/com/keylesspalace/tusky/components/login/LoginActivity.kt

```diff
@@ -166,32 +165,33 @@ class LoginActivity : BaseActivity(), Injectable {
         setLoading(true)
 
         lifecycleScope.launch {
-            val credentials: AppCredentials = try {
-                mastodonApi.authenticateApp(
-                    domain, getString(R.string.app_name), oauthRedirectUri,
-                    OAUTH_SCOPES, getString(R.string.tusky_website)
-                )
-            } catch (e: Exception) {
-                binding.loginButton.isEnabled = true
-                binding.domainTextInputLayout.error =
-                    getString(R.string.error_failed_app_registration)
-                setLoading(false)
-                Log.e(TAG, Log.getStackTraceString(e))
-                return@launch
-            }
-
-            // Before we open browser page we save the data.
-            // Even if we don't open other apps user may go to password manager or somewhere else
-            // and we will need to pick up the process where we left off.
-            // Alternatively we could pass it all as part of the intent and receive it back
-            // but it is a bit of a workaround.
-            preferences.edit()
-                .putString(DOMAIN, domain)
-                .putString(CLIENT_ID, credentials.clientId)
-                .putString(CLIENT_SECRET, credentials.clientSecret)
-                .apply()
-
-            redirectUserToAuthorizeAndLogin(domain, credentials.clientId)
+            mastodonApi.authenticateApp(
+                domain, getString(R.string.app_name), oauthRedirectUri,
+                OAUTH_SCOPES, getString(R.string.tusky_website)
+            ).fold(
```

Comment:
I must say I don't like using `fold` except for one very specific case: when you assign to a variable and provide default value. I would argue that `try/catch` is there for a reason.

Reply:
But `fold` returns a value, you can do something like

```
val data  = mastodonApi.getData().fold(
    { // map data
    },
    { // return default value
    }
)
```

If you just want a default value you can do 

```
val data = mastodonApi.getData().getOrDefault(/* whatever default you need */)
```

Reply:
Ooooh I got some language soup in my head, I'm sorry :sweat_smile: 
Of course it's `fold()`, it's a `Result`! 
Don't listen to me, everything is great.